### PR TITLE
Odds and ends improvements

### PR DESCRIPTION
**Pull request by josharian**
I'm continuing to upstream my local go-fuzz changes. This is another batch of little stuff.

### CODE REVIEW

File: go-fuzz/mutator.go [hunk 1]
```diff
@@ -153,59 +160,41 @@ func (m *Mutator) mutate(data []byte, ro *ROData) []byte {
...
-			case 1:
-				*(*uint16)(unsafe.Pointer(&res[pos])) -= v
-			case 2:
-				x := *(*uint16)(unsafe.Pointer(&res[pos]))
-				*(*uint16)(unsafe.Pointer(&res[pos])) = swap16(swap16(x) + v)
-			case 3:
-				x := *(*uint16)(unsafe.Pointer(&res[pos]))
-				*(*uint16)(unsafe.Pointer(&res[pos])) = swap16(swap16(x) - v)
+			if m.r.Bool() {
+				v = ^(v - 1) // v *= -1, but for uints
```
Comment (dvyukov) [inline go-fuzz/mutator.go]: Won't `v = 0 - v` be more readable?

File: go-fuzz/mutator.go [hunk 2]
```diff
@@ -221,23 +210,19 @@ func (m *Mutator) mutate(data []byte, ro *ROData) []byte {
...
 				continue
 			}
 			pos := m.rand(len(res) - 3)
+			buf := res[pos:]
 			v := uint32(interesting32[m.rand(len(interesting32))])
-			if m.r.Bool() {
-				v = swap32(v)
-			}
-			*(*uint32)(unsafe.Pointer(&res[pos])) = v
+			m.randByteOrder().PutUint32(buf, v)
```
Comment (dvyukov) [inline go-fuzz/mutator.go]: How does this affect performance? I have a memory that encoding/binary is slow.

File: go-fuzz/mutator.go (hunk 1 above)
Comment (josharian) [reply go-fuzz/mutator.go]: Indeed, thanks. Will do.

File: go-fuzz/mutator.go (hunk 2 above)
Comment (josharian) [reply go-fuzz/mutator.go]: Negligible impact. encoding/binary is fast now: The compiler recognizes the load pattern and converts it to a single unaligned load, equivalent to the unsafe magic, at least on amd64, which is where (I imagine) the vast majority of all fuzzing occurs. And the unsafe variant can crash on architectures that don't support unaligned loads; encoding/binary will at least be correct and slow. :)

Comment (josharian) [reply go-fuzz/mutator.go]: To be precise: LittleEndian loads are converted to a single unaligned load. BigEndian loads are converted to a single unaligned load and a BSWAPQ.

**Pull request merged**